File: src/main/java/org/example/kalkulationsprogramm/repository/KundeNotizRepository.java
package org.example.kalkulationsprogramm.repository;

import org.example.kalkulationsprogramm.domain.KundeNotiz;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.List;

@Repository
public interface KundeNotizRepository extends JpaRepository<KundeNotiz, Long> {

    List<KundeNotiz> findByKundeIdOrderByErstelltAmDesc(Long kundeId);

    List<KundeNotiz> findByKundeIdAndTextContainingIgnoreCaseOrderByErstelltAmDesc(
            Long kundeId, String query);
}

File: src/main/kotlin/org/example/kalkulationsprogramm/repository/KundeNotizRepository.kt
package org.example.kalkulationsprogramm.repository

import org.example.kalkulationsprogramm.domain.KundeNotiz
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

@Repository
interface KundeNotizRepository : JpaRepository<KundeNotiz, Long> {
    fun findByKundeIdOrderByErstelltAmDesc(kundeId: Long?): List<KundeNotiz>

    fun findByKundeIdAndTextContainingIgnoreCaseOrderByErstelltAmDesc(
        kundeId: Long?,
        query: String?,
    ): List<KundeNotiz>
}
